src/dove/core/transfers.py

```diff
@@ -25,63 +25,80 @@
-
 if TYPE_CHECKING:
     from . import Resource
 
 TransferFunc: TypeAlias = "RatioTransfer | PolynomialTransfer"
 
 
-@dataclass
+@dataclass()
 class RatioTransfer:
     """
-    A transfer class that enforces a ratio relationship between input and output resources.
+    A transfer class that enforces ratio relationships between multiple input and output resources.
 
-    This class models the conversion of one resource to another with a specified ratio.
-    For example, it could represent energy conversion efficiency or material transformation.
+    This class models the conversion of one or more resources to one or more other resources
+    according to specified ratios between each. For example, it could represent energy conversion
+    with less than 100% efficiency and multiple required inputs or material transformation that
+    involves relevant waste products. This class is intended to be used to enforce relationships
+    between input and output quantities as well as ratios between multiple different inputs and
+    ratios between multiple different outputs. This is comparable to a chemical balance
+    relationship, where A, B, C, and D are constant floats:
+        A(input_resource_1) + B(input_resource_2) -> C(output_resource_1) + D(output_resource_2)
 
     Parameters
     ----------
-    input_res : Resource
-        The input resource object.
-    output_res : Resource
-        The output resource object.
-    ratio : float, default=1.0
-        The conversion ratio from input to output.
-        A ratio of 1.0 means the output equals the input.
-        A ratio of 0.9 means 90% of input becomes output (10% loss).
-        A ratio > 1.0 means the output is amplified relative to input.
+    input_resources : dict[Resource, float]
+        A dictionary keyed by the input resources to the component with values that are the
+        relative amounts required of that resource to enable conversion to output resources.
+    output_resources : Resource
+        A dictionary keyed by the output resources to the component with values that are the
+        relative amounts produced of that resource when the inputs are supplied.
+
 
     Examples
     --------
-    Create a transfer that converts electricity to heat with 95% efficiency:
+    Create a transfer that converts heat to electricity at 90% efficiency (assuming "heat" and
+    "electricity" are both Resource instances and are both in MW):
+
+    >>> transfer = RatioTransfer(
+            input_resources={heat: 1.0},
+            output_resources={electricity: 0.9},
+        )
+
+    Later, after component instantiation, the transfer can be used to evaluate a
+    constraint with dispatch activity values:
+
+    >>> transfer(inputs={"heat": 200}, outputs={"electricity": 180})
+    [200, 200]
+
+    Since these values are all equal, the constraint should be satisfied.
+
+    Create a transfer that converts 1.8 units of heat and 1.0 units of electricity to 0.5 units of
```

Comment:
if we do keep this section (see the comment from line 72), might be important to include a comment that these are fictional values (unless you have a citation for them). I think the heat/electricity example is fine, but I worry that a user might see these hydrogen conversion ratios and take them as fact